proto/src/determined/resourcepool/v1/resourcepool.proto

```diff
@@ -0,0 +1,315 @@
+  int32 num_agents = 4;
+  // The total number of slots that exist in the resource pool
+  int32 slots_available = 5;
+  // The number of slots that are actively running workloads
+  int32 slots_used=6;
+  // The max number of CPU containers that can run in this resource pool
+  int32 cpu_container_capacity = 7;
+  // The current number of CPU containers running in this resource pool
+  int32 cpu_containers_running = 8;
+  // Is this resource pool the default GPU pool?
+  bool default_gpu_pool=9;
+  // Is this resource pool the default CPU pool?
+  bool default_cpu_pool=10;
+  // Is this resource pool using preemptible/spot instances? Only meaningful in an AWS or GCP resource pool.
+  bool preemptible=11;
+  // When using dynamic agents, the minimum number of agents that can exist in the resource pool.
+  int32 min_agents=12;
+  // When using dynamic agents, the maximum number of agents that can exist in the resource pool.
+  int32 max_agents=13;
+  // The maximum number of CPU containers that can run on an individual agent
+  int32 cpu_container_capacity_per_agent=15;
+  // The type of the scheduler. Either 'FAIR_SHARE', 'PRIORITY', or 'ROUND_ROBIN'
+  SchedulerType scheduler_type=16;
+  // The fitting policy of the scheduler.
+  FittingPolicy scheduler_fitting_policy=17;
+  // The location of the resource pool. For AWS this returns the region and for GCP this return the zone.
+  string location=18;
+  // The VM image used for the agents when using dynamic agents.
+  string image_id=19;
+  // The instance type of the agents when using dynamic agents. For AWS this is the Instance Type. For GCP
+  // this is the machine type combined with the number and types of GPUs. To work with this data
+  // programattically, we recommend working with the ResourcePool.details.aws.instanceType and
+  // ResourcePool.details.gcp.machineType/gpuType/gpuNum.
+  string instance_type=20;
+
+  // The url of the Determined master
+  string master_url=21;
+  // A hostname for which the master’s TLS certificate is valid, if the host specified
+  // by the master_url option is an IP address or is not contained in the certificate
+  string master_cert_name=22;
+  // The startup script for the agent. This runs on the node the agent runs on.
+  string startup_script=23;
+  // The startup script for the agent's container. This runs in the container determined-agent runs in.
+  string container_startup_script=24;
+  // The Docker network to use for the agent when using dynamic agents.
+  string agent_docker_network=25;
+  // The docker runtime to use for the agent when using dynamic agents
+  string agent_docker_runtime=26;
+  // The docker image to use for the agent when using dynamic agents
+  string agent_docker_image=27;
+  // The Fluent docker image to use
+  string agent_fluent_image=28;
+  // The maximum idle period of agents. The master waits for this period of time
+  // before shutting down idle agents.
+  string max_idle_agent_period=29;
```

Comment:
question: what's the unit here? is this numerical? 
same for  max_agent_starting_period

Reply:
i see it's returned as `2H3m`

Reply:
I think we should leave it as a string. We could convert it to a number if we make a decision on the granularity (seconds, etc), but while that would be close to correct, we technically could be not quite correct. And I don't think the client should need to do anything other than display it as a string. 

If we have a use case where the client needs it to be a number, we can change this and have the potential truncation issue being a documented quirk of the API.

Reply:
what's the max period or duration we might see? int32 or uint32 could handle a lot in seconds. IMO number would give the client the most flexibility in how they'd want to display duration although the format here looks very reasonable. Another point being sorting, and if the client wants to sort by this field clientside.
It works either way for me. :+1:

Reply:
if we get it as a string, we can only display it straight up. Whereas if we get it as a number in seconds, it gives us flexibility on how to display it and potentially perform math on it (like if we wanted to add up different durations).

Reply:
Alright, I converted it to a float number of seconds